src/main/java/net/dv8tion/jda/core/entities/impl/WebhookImpl.java

```diff
@@ -0,0 +1,192 @@
+
+
+    public WebhookImpl(TextChannel channel, String id)
+    {
+        this.channel = channel;
+        this.id = id;
+    }
+
+    @Override
+    public JDA getJDA()
+    {
+        return channel.getJDA();
+    }
+
+    @Override
+    public Guild getGuild()
+    {
+        return channel.getGuild();
+    }
+
+    @Override
+    public TextChannel getChannel()
+    {
+        return channel;
+    }
+
+    @Override
+    public Member getOwner()
+    {
+        return owner;
+    }
+
+    @Override
+    public User getDefaultUser()
+    {
+        return user;
+    }
+
+    @Override
+    public String getName()
+    {
+        return user.getName();
+    }
+
+    @Override
+    public String getToken()
+    {
+        return token;
+    }
+
+    @Override
+    public String getUrl()
+    {
+        return "https://discordapp.com/api/webhooks/" + getId() + "/" + getToken();
+    }
+
+    @Override
+    public RestAction<Void> delete()
+    {
+        Route.CompiledRoute route = Route.Webhooks.DELETE_TOKEN_WEBHOOK.compile(id, token);
+        return new RestAction<Void>(getJDA(), route, null)
+        {
+            @Override
+            protected void handleResponse(Response response, Request request)
+            {
+                if (response.isOk())
+                    request.onSuccess(null);
+                else
+                    request.onFailure(response);
+            }
+        };
+    }
+
+    @Override
+    public WebhookManager getManager()
+    {
+        WebhookManager mng = manager;
+        if (mng == null)
+        {
+            synchronized (mngLock)
+            {
+                mng = manager;
+                if (mng == null)
+                    mng = manager = new WebhookManager(this);
+            }
+        }
+        return mng;
+    }
+
+    @Override
+    public WebhookManagerUpdatable getManagerUpdatable()
+    {
+        WebhookManagerUpdatable mng = managerUpdatable;
+        if (mng == null)
+        {
+            synchronized (mngLock)
+            {
+                mng = managerUpdatable;
+                if (mng == null)
+                    mng = managerUpdatable = new WebhookManagerUpdatable(this);
+            }
+        }
+        return mng;
+    }
+
+    @Override
+    public String getId()
+    {
+        return id;
+    }
+
+    /* -- Impl Setters -- */
+
+    public WebhookImpl setOwner(Member member)
+    {
+        this.owner = member;
+        return this;
+    }
+
+    public WebhookImpl setToken(String token)
+    {
+        this.token = token;
+        return this;
+    }
+
+    public WebhookImpl setUser(User user)
+    {
+        this.user = user;
+        return this;
+    }
+
+    /* -- Object Overrides -- */
+
+    @Override
+    public int hashCode()
+    {
+        return getId().hashCode();
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        return obj instanceof ISnowflake
+                && ((Webhook) obj).getId().equals(id);
```

Comment:
Check if it is specifically an instance of Webhook. Although *technically* it should be impossible for them to share the same id, you should still always check the proper type, especially since you cast to it.